src/plane.rs

```diff
@@ -188,52 +126,22 @@ impl<T: Pixel> PlaneData<T> {
     cfg_if::cfg_if! {
       if #[cfg(target_arch = "wasm32")] {
         // FIXME: wasm32 allocator fails for alignment larger than 3
-        const DATA_ALIGNMENT_LOG2: usize = 3;
+        const DATA_ALIGNMENT: usize = 1 << 3;
       } else {
-        const DATA_ALIGNMENT_LOG2: usize = 6;
+        const DATA_ALIGNMENT: usize = 1 << 6;
       }
     }
 
-    unsafe fn layout(len: usize) -> Layout {
-        Layout::from_size_align(len * mem::size_of::<T>(), 1 << Self::DATA_ALIGNMENT_LOG2)
-            .expect("layout size too large")
-    }
-
-    unsafe fn new_uninitialized(len: usize) -> Self {
-        let ptr = {
-            let layout = Self::layout(len);
-            let ptr = alloc(layout) as *mut T;
-            if ptr.is_null() {
-                handle_alloc_error(layout);
-            }
-            std::ptr::NonNull::new_unchecked(ptr)
-        };
-
-        PlaneData {
-            ptr,
-            len,
-            _marker: PhantomData,
-        }
-    }
-
     pub fn new(len: usize) -> Self {
-        // SAFETY: we initialize the plane data before returning
-        let mut pd = unsafe { Self::new_uninitialized(len) };
-
-        for v in pd.iter_mut() {
-            *v = T::cast_from(128);
+        Self {
+            data: avec_rt!([Self::DATA_ALIGNMENT]| T::cast_from(0); len),
```

Comment:
This should probably not use `avec_rt` because it's technically a runtime-variable alignment. You could use `AVec::<T, ConstAlign<Self::DATA_ALIGNMENT>>` and manually create it. Bit less convenient but carries a compile-time alignment guarantee.

Reply:
we can use from_iter and keep in mind we want `128` as value.

Reply:
Yeah the downside of the aligned_vec crate is that nothing is documented. My assumption is that because we needed the alignment to be different on wasm, we needed to use a runtime alignment here, but I suppose we could do that in a different way and still use const align.

Reply:
It is at compile time anyway, so it is a matter of using `type A = ConstAlign<DATA_ALIGNMENT>` and then use it where needed.